SharedUtils/src/main/java/com/braintreepayments/api/HttpNoResponse.java

```diff
@@ -1,12 +0,0 @@
-package com.braintreepayments.api;
-
-class HttpNoResponse implements HttpResponseCallback {
-
-    @Override
-    public void success(String responseBody) {
```

Comment:
Why we opting now for `onResult` instead of `success` and `failure` callbacks?

Reply:
We use the [Single Abstract Method](https://www.javacodemonk.com/single-abstract-method-sam-and-functional-interface-in-java-44f22fc3#:~:text=Interface%20that%20has%20single%20abstract,has%20exactly%20one%20abstract%20method.) pattern throughout the sdk. We deferred this change for a while since it was internal. SAM is really good for Kotlin interop though, if we end up converting more classes to Kotlin in the future this will make it a lot easier.

Reply:
Awesome - thank you for this context!

Reply:
🙏